### ISSUE TITLE
Rule Change: (max-params)

### ISSUE THREAD

**Issue by GP4cK** [eslint/eslint#15745]
### What rule do you want to change?

max-params

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
@Injectable()
export class TodoService {
  constructor(
    @Inject(REQUEST) private readonly request: AugmentedRequest,
    @InjectRepository(Project) private readonly projectRepo: Repository<Project>,
    @InjectRepository(Todo) private readonly todoRepo: Repository<Todo>,
    @InjectRepository(User) private readonly userRepo: Repository<User>,
  ) {}
```

### What does the rule currently do for this code?

It generates a warning: "Constructor has too many parameters".

### What will the rule do after it's changed?

If we had an option to handle constructors differently than functions/methods, we could get rid of this warning.

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

I'm reopening this issue https://github.com/eslint/eslint/issues/7392.
I'm using [NestJS](https://nestjs.com/) which relies heavily on Dependency Injection. I'm happy to restrict the functions and methods to a limited number of parameters but I often have to create Services which require a bit more parameters... I've also tried to use this [plugin](https://www.npmjs.com/package/eslint-plugin-max-params-no-constructor) but it doesn't override the max-params rule.

**Comment by nzakas** [comment]
Thanks for the input. As [stated in the README](https://github.com/eslint/eslint#stylistic-rule-updates), we actually aren't making any further updates to rules that enforce stylistic preferences. If you have a strong preference for such a change, you can always copy the existing rule into a custom rule and make it behave the way you want.

**Comment by GP4cK** [comment]
> we actually aren't making any further updates to rules that enforce stylistic preferences

Sorry I missed this.
I ended up creating a small plugin: https://www.npmjs.com/package/eslint-plugin-better-max-params.
I hope it helps someone else.
